### ISSUE TITLE
Visualize domain decomposition in Paraview

### ISSUE THREAD

**Issue by tangqi** [mfem/mfem#1733]
Hi,
Is there a simple way to visualize domain decomposition in the current paraview format? This should be a simple task but I could not find a solution.

The reason I am asking is I am doing a lot of production runs with AMR and dynamic load-balancing. So it would be good to see domain decomposition. Paraview works best for me since that's the only tool I got the client-server mode working. But if that's not easy, I am also open to any other suggestions. Glvis is very slow to read data files saved in thousands of cores on my HPCs. 

Thanks,
Qi

**Comment by pazner** [comment]
By domain decomposition, do you mean to see which MPI rank is associated with each element?

**Comment by tangqi** [comment]
Yes, I want to see MPI ranks for each element. I have tried something like "processorid" but I can only visualize the MPI ranks associated with the processors I used for visualization jobs. 

PS, I was told by a dealii developer that they have to save that part of information explicitly.

**Comment by pazner** [comment]
Yes, ParaView will not automatically show that information. However, you can easily save it yourself using a p=0 DG field.

For example, in example 9p, I can do the following:

```c++
L2_FECollection pw_const_fec(0, dim);
ParFiniteElementSpace pw_const_fes(pmesh, &pw_const_fec);
ParGridFunction mpi_rank_gf(&pw_const_fes);
mpi_rank_gf = myid;
```
and then
```c++
pd->RegisterField("MPI rank", &mpi_rank_gf);
```

**Comment by tangqi** [comment]
Ah. That's a good idea. I will give it a try. Thanks a lot!

**Comment by v-dobrev** [comment]
Isn't the data saved to different files for different processors? I'd assume that this decomposition (corresponding to the files) is something you get access to in ParaView without the need to explicitly save it. At least in VisIt, I think that is available, but I may be wrong.

**Comment by pazner** [comment]
The data is somehow "available" based on the partitioning into files, but it's not easily accessible/visualizable in ParaView without explicitly saving it as a field.

**Comment by tzanio** [comment]
@tangqi, any cool pictures?

**Comment by tangqi** [comment]
We were able to capture the multiscale behavior with dynamic AMR and a relatively large time step. The domain decomposition plot looks pretty good using the trick @pazner suggested. 

(I am new to paraview, so still need to improve the paraview skills. But paraview is the best solution I found so far with wfh.)

![dd](https://user-images.githubusercontent.com/1406660/91916887-8e415400-ec7b-11ea-9c64-003256b0892e.png)
![current](https://user-images.githubusercontent.com/1406660/91916893-939e9e80-ec7b-11ea-94bf-b12dde80ff54.png)
